taxonomy/tests/test_api.py

```diff
@@ -0,0 +1,328 @@
+
+        self.learning_resource.terms.add(self.term)
+        self.assertEquals([self.learning_resource],
+                          list(get_learning_resources_for_term(self.term.id)))
+
+        self.assertRaises(Http404,
+                          lambda: get_learning_resources_for_term(9))
+
+    def test_get_term(self):
+        """
+        Test get_term
+        """
+        invalid_id = -1
+        self.assertRaises(Http404, lambda: get_term(invalid_id))
+        self.assertEquals(self.term, get_term(self.term.id))
+
+    def test_get_terms_for_learning_resource(self):
+        """
+        Test get_terms_for_learning_resource
+        """
+        self.assertEquals([],
+                          list(get_terms_for_learning_resource(
+                              self.learning_resource.id)))
+
+        self.term.learning_resources.add(self.learning_resource)
+        self.assertEquals([self.term],
+                          list(get_terms_for_learning_resource(
+                              self.learning_resource.id)))
+        self.assertRaises(Http404,
+                          lambda: get_terms_for_learning_resource(9))
+
+    def test_get_types_for_vocabulary(self):
+        """
+        Test get_types_for_vocabulary
+        """
+        self.vocabulary.learning_resource_types.add(
+            self.learning_resource_type1)
+        self.assertEquals([self.learning_resource_type1.name],
+                          list(get_types_for_vocabulary(self.vocabulary.id)))
+        self.assertRaises(Http404,
+                          lambda: get_types_for_vocabulary(3))
+
+    def test_get_vocabularies_for_type(self):
+        """
+        Test get_vocabularies_for_type
+        """
+        self.vocabulary.learning_resource_types.add(
+            self.learning_resource_type1)
+        self.assertEquals([self.vocabulary],
+                          list(get_vocabularies_for_type(
+                              self.learning_resource_type1.name)))
+        self.assertRaises(Http404,
+                          lambda: get_vocabularies_for_type("xyz"))
+
+    def test_get_vocabulary(self):
+        """
+        Test get_vocabulary
+        """
+
+        actual_vocabulary = get_vocabulary(self.vocabulary.id)
+        self.assertEquals(self.vocabulary, actual_vocabulary)
+
+        self.assertRaises(Http404,
+                          lambda: get_vocabulary(3))
+        self.assertRaises(Http404,
+                          lambda: get_vocabulary(None))
+
+    def test_remove_term_from_learning_resource(self):
+        """
+        Test remove_term_from_learning_resource
+        """
+        self.learning_resource.terms.add(self.term)
+        self.assertEquals([self.term],
+                          list(get_terms_for_learning_resource(
+                              self.learning_resource.id)))
+        remove_term_from_learning_resource(self.learning_resource.id,
```

Comment:
This fixes the other bug. The arguments here were reversed (it was sending term ID then learning resource ID, but the function takes them in the other order). This probably doesn't fail in certain circumstances if they're both the same, probably `1`.